### ISSUE TITLE
How to update weights of CodeBERT for a downstream task? 

### ISSUE THREAD

**Issue by dashpritam** [microsoft/CodeBERT#190]
I'm using CodeBERT for a downstream multi-class classification task. I'm training `RobertaForSequenceClassification` model, following #53. But this only updates the weights of the classification layer, but not the weights of CodeBERT layers. 

I'm interested to evaluate if updating CodeBERT's weights (slightly) for a downstream task improves the results. Please share your thoughts.

**Comment by guoday** [comment]
The code updates the weights of the whole CodeBERT instead of only classification layer

**Comment by dashpritam** [comment]
Thank you for your response. I'll look into my code and check why CodeBERT's weights are not updating. I finetuned `RobertaForSequenceClassification ` for 5 epochs and in none of the epochs, CodeBERT's weights are updating. 

Another related question - how is the loss calculated below? This is from finetuning code in #53 
logits - (768,) CLS token
labels - [1, 0, 1, 0, .., 0] (one hot encode or class label for multiclass classification)

wouldn't logits and labels have different shapes? 
```
def forward(self, input_ids=None,labels=None): 
        logits=self.encoder(input_ids,attention_mask=input_ids.ne(1))[0]
        prob=torch.softmax(logits,-1)
        if labels is not None:
            loss_fct = nn.CrossEntropyLoss(ignore_index=-1)
            loss = loss_fct(logits,labels)
            return loss,prob
        else:
            return prob 
```

**Comment by guoday** [comment]
1. Normally, CodeBERT is updated. You can use the following code to print which parameters are updated.
`    for n, p in model.named_parameters():
        if p.requires_grad:
            print(n)
`
2.  `logits` are (bs,labels_num) and `labels` are label index which shape is (bs,)
